src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/EventServiceImpl.java

```diff
@@ -146,37 +119,64 @@ private void logAndThrow(final CollectionExercise collectionExercise) throws CTP
-    if (collex != null) {
-      Event event = this.eventRepository.findOneByCollectionExerciseAndTag(collex, tag);
-      if (event != null) {
-        event.setTimestamp(new Timestamp(date.getTime()));
+  public Event updateEvent(final UUID collexUuid, final String tag, final Date date)
+      throws CTPException {
+    final CollectionExercise collex = getCollectionExercise(collexUuid, Fault.BAD_REQUEST);
+    final Event event = getEventByTagAndCollectionExerciseId(tag, collex);
 
-        List<Event> existingEvents = this.eventRepository.findByCollectionExercise(collex);
+    event.setTimestamp(new Timestamp(date.getTime()));
+    validateUpdatedEvents(collex, event);
+    updateActionRules(collex, event);
 
-        if (this.eventValidator.validate(existingEvents, event, collex.getState())) {
+    eventRepository.save(event);
 
-          this.eventRepository.save(event);
+    fireEventChangeHandlers(MessageType.EventUpdated, event);
 
-          fireEventChangeHandlers(MessageType.EventUpdated, event);
-        } else {
-          throw new CTPException(
-              CTPException.Fault.BAD_REQUEST, String.format("Invalid event update"));
-        }
+    return event;
+  }
 
-      } else {
-        throw new CTPException(
-            CTPException.Fault.RESOURCE_NOT_FOUND,
-            String.format("Event %s does not exist", event.getId()));
-      }
+  private void updateActionRules(final CollectionExercise collex, final Event event)
+      throws CTPException {
+    final CaseTypeOverride bCaseOverride =
+        caseTypeOverrideService.getCaseTypeOverride(collex, BUSINESS_SAMPLE_UNIT_TYPE);
+    final CaseTypeOverride biCaseOverride =
+        caseTypeOverrideService.getCaseTypeOverride(collex, BUSINESS_INDIVIDUAL_SAMPLE_UNIT_TYPE);
+    final SurveyDTO survey = surveySvcClient.getSurveyForCollectionExercise(collex);
+
+    for (final ActionRuleUpdater aru : actionRuleUpdaters) {
+      aru.execute(event, bCaseOverride, biCaseOverride, survey);
+    }
+  }
 
-      return event;
-    } else {
+  private void validateUpdatedEvents(final CollectionExercise collex, final Event event)
+      throws CTPException {
+    final List<Event> existingEvents = eventRepository.findByCollectionExercise(collex);
+
+    if (!eventValidator.validate(existingEvents, event, collex.getState())) {
+      throw new CTPException(Fault.BAD_REQUEST, "Invalid event update");
+    }
+  }
+
+  private Event getEventByTagAndCollectionExerciseId(
+      final String tag, final CollectionExercise collex) throws CTPException {
+    final Event event = eventRepository.findOneByCollectionExerciseAndTag(collex, tag);
+    if (event == null) {
+      throw new CTPException(
+          Fault.RESOURCE_NOT_FOUND,
+          String.format("Event %s for Collection Exercise %s does not exist", tag, collex.getId()));
+    }
+    return event;
+  }
+
+  private CollectionExercise getCollectionExercise(final UUID collexUuid, final Fault badRequest)
```

Comment:
maybe change the variable name if it isn't always going to be a BAD_REQUEST fault?